File: CSharp/SistemaAcademico/Models/Materia.cs
using System.ComponentModel.DataAnnotations;

namespace SistemaAcademico.Models;
public class Materia {
    public int Id { get; set; }
    public required string Nombre { get; set; }
    
    // Clave Foránea automática
    public int ProfesorId { get; set; }
    public Profesor Profesor { get; set; } = null!;

    // Relación: Una materia tiene muchos estudiantes
    public List<Estudiante> Estudiantes { get; set; } = [];
}

File: PHP/mi_proyecto/app/Models/Materia.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Materia extends Model
{
    protected $fillable = ['nombre', 'codigo', 'profesor_id'];

    // Relación: Una materia pertenece a un profesor
    public function profesor(): BelongsTo
    {
        return $this->belongsTo(Profesor::class, 'profesor_id');
    }

    // Relación: Una materia pertenece a muchos estudiantes (Inversa de la relación N:N)
    public function estudiantes(): BelongsToMany
    {
        return $this->belongsToMany(Estudiante::class, 'estudiante_materia');
    }
}